src/wallet/injected/wallet-connection-sender.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    RequestSignTransactionsOptions,
+    SignInOptions,
+    WalletConnection,
+} from "../wallet-connection";
+
+import { Near } from '../../near';
+import { Account } from '../../account'
+import { SenderWallet } from "./wallet-injected-interface";
+
+export class WalletConnectionSender extends WalletConnection {
+    private senderWallet: SenderWallet;
+
+
+    constructor(near: Near, appKeyPrefix: string | null, walletName: string) {
+        console.warn("This implementation is temporary, regular Injected Wallet should be used instead.");
+        super(near, appKeyPrefix);
+        this.senderWallet = window[walletName] as SenderWallet;
+    }
+
+    async requestSignTransactions({ transactions, meta, callbackUrl }: RequestSignTransactionsOptions): Promise<void> {
+        this.senderWallet.requestSignTransactions({ transactions })
+            .then((res) => {
+                console.log('requestSignTransactions res:', res);
+                return res;
+            });
+    }
+    async requestSignIn({ contractId, methodNames, successUrl, failureUrl }: SignInOptions): Promise<void> {
+        const { accessKey } = await this.senderWallet.requestSignIn({
+            contractId: contractId,
+        });
+        //TODO: why do we need accessKey here?
+        console.log("New function call key added:", accessKey);
```

Comment:
I think the purpose is to make the function call key accessible to DApp that wants to sign transaction using `near-api-js`, but the way to make this work could be improved.